### PR TITLE
'self' as pseudo-identifier in resource paths

### PR DESCRIPTION
**Pull request by tfrauenstein**
fixes https://github.com/zalando/restful-api-guidelines/issues/595

### CODE REVIEW

File: chapters/resources.adoc [hunk 1]
```diff
@@ -85,7 +85,7 @@ consisting of <<241, compound keys>>.
 Some API resources may contain or reference sub-resources. Embedded
 sub-resources, which are not top-level resources, are parts of a higher-level
 resource and cannot be used outside of its scope. Sub-resources should be
-referenced by their name and identifier in the path segments as follows:
+referenced by their identifier and name in the path segments as follows:
```
Comment (tkrop) [inline chapters/resources.adoc:88]: Not sure about this change: Logical order (of appearance)  is sub-resource name and identifier, not reverse... but may be I'm to literal.

File: chapters/resources.adoc [hunk 2]
```diff
@@ -94,22 +94,28 @@ referenced by their name and identifier in the path segments as follows:
...
-/customers/12ev123bv12v
 /shopping-carts/de:1681e6b88ec1/items/1
 /shopping-carts/de:1681e6b88ec1
 /content/images/9cacb4d8
 /content/images
 ----
 
 **Note:** resource identifiers may be build of multiple other resource
-identifiers (see <<241>>).
+identifiers -- see <<241>>.
```
Comment (tkrop) [inline chapters/resources.adoc:110]: We should not inconsistently start to change reference patterns from `(see ...)` to `-- see ...`.

File: chapters/resources.adoc [hunk 3]
```diff
@@ -94,22 +94,28 @@ referenced by their name and identifier in the path segments as follows:
...
+
+**Exception:** In some situations the id information is not passed
```
Comment (tkrop) [inline chapters/resources.adoc:112]: May I suggest to substitute `the id information` against `the resource identifier`? Else I think it should be `the ID information`.

File: chapters/resources.adoc [hunk 4]
```diff
@@ -94,22 +94,28 @@ referenced by their name and identifier in the path segments as follows:
...
 /content/images/9cacb4d8
 /content/images
 ----
 
 **Note:** resource identifiers may be build of multiple other resource
-identifiers (see <<241>>).
+identifiers -- see <<241>>.
+
+**Exception:** In some situations the id information is not passed 
+as a path segment but via context information, e.g. the authorization token.
```
Comment (tkrop) [inline chapters/resources.adoc:113]: Should we really be open to all context information? `self` in my opinion is restricted to `authorization information`. So may be we should just write: `but via authorization information, e.g. an authorization token or session cookie.`

File: chapters/naming.adoc [hunk 5]
```diff
@@ -129,9 +129,11 @@ on HTTP headers.
 == {MUST} pluralize resource names
 
 Usually, a collection of resource instances is provided (at least API
-should be ready here). The special case of a resource singleton is a
-collection with cardinality 1.
+should be ready here). The special case of a _resource singleton_ is a
+collection with cardinality 1 --  to be highlighted in the resource
```
Comment (tkrop) [inline chapters/naming.adoc:133]: Not sure what to be highlighted in the specification here? I think what we should say here is:

```
The special case of a _resource singleton_ must be modeled as a collection with cardinality 1.

**Exception:** the _pseudo identifier_ `self` used to specify a resource endpoint where the
resource identifier provided by authorization information (see <<143>>).
```

File: chapters/resources.adoc [hunk 6]
```diff
@@ -94,22 +94,28 @@ referenced by their name and identifier in the path segments as follows:
...
 ----
 
 **Note:** resource identifiers may be build of multiple other resource
-identifiers (see <<241>>).
+identifiers -- see <<241>>.
+
+**Exception:** In some situations the id information is not passed 
+as a path segment but via context information, e.g. the authorization token. 
+Here, it is reasonable to use **`self`** as _pseudo-identifier_ path segment. 
+For instance, you may define `/employee/self` or `/employee/self/personal-details`
```
Comment (tkrop) [inline chapters/resources.adoc:115]: I think this should not be `/employee/self` but `/employees/self` to be consistent with rules and allow for a compatible update to `employees/{id}`.

File: chapters/naming.adoc (hunk 5 above)
Comment (tfrauenstein) [reply chapters/naming.adoc:133]: ok, and will add hint to explicitly define the cardinality constraint.

File: chapters/naming.adoc [hunk 7]
```diff
@@ -128,9 +128,14 @@ on HTTP headers.
...
-should be ready here). The special case of a resource singleton is a
-collection with cardinality 1.
+Usually, a collection of resource instances is provided (at least the API 
+should be ready here). The special case of a _resource singleton_ must 
+be modeled as a collection with cardinality 1 including definition of 
+`maxItems` = `minItems` = 1 for the returned `array` structure 
+to make the cardinality constraint explicit.
+
+**Exception:** the _pseudo identifier_ `self` used to specify a resource endpoint 
+where the resource identifier provided by authorization information (see <<143>>).
```
Comment (tkrop) [inline chapters/naming.adoc:138]: `provided` -> `is provided`

Comment (tfrauenstein) [reply chapters/naming.adoc:138]: 👍

**Comment by tkrop** [comment]
:+1:

**Comment by ePaul** [comment]
:+1:

**Pull request merged**